devito/types/dimension.py

```diff
@@ -898,39 +901,52 @@ class ConditionalDimension(DerivedDimension):
     is_NonlinearDerived = True
     is_Conditional = True
 
-    __rkwargs__ = DerivedDimension.__rkwargs__ + ('factor', 'condition', 'indirect')
+    __rkwargs__ = DerivedDimension.__rkwargs__ + \
+        ('symbolic_factor', 'factor', 'condition', 'indirect')
 
     def __init_finalize__(self, name, parent=None, factor=None, condition=None,
-                          indirect=False, **kwargs):
+                          indirect=False, symbolic_factor=None, **kwargs):
```

Comment:
Docstring needs updating since `symbolic_factor` now included in kwargs. How is `symbolic_factor` different to `factor`?

Reply:
Makes sense. Does it `_rebuild` correctly? I seem to remember that there is still an issue where rebuilding a `ConditionalDimension` with no factor introduces a factor of one unless that got patched when I wasn't looking

Reply:
This rebuild issue been patch a while back yes